cursive-core/src/views/dialog.rs

```diff
@@ -167,6 +168,19 @@ impl Dialog {
         self.invalidate();
     }
 
+    /// Replace the content for this dialog.
+    ///
+    /// Previous content will be returned as a `BoxedView`.
+    pub fn replace_content<V: IntoBoxedView>(&mut self, view: V) -> BoxedView {
+        let old_content = replace(
+            &mut self.content,
+            LastSizeView::new(BoxedView::boxed(view)),
+        );
+        self.invalidate();
```

Comment:
We could first invalidate, then replace the content (so no need for a temp variable).
We also may want to call `BoxedView::unwrap` on the result and return a `Box<dyn View>` - this is what `StackView` does currently when poping layers.

We could also do that in the `set_content` method, we might not need a dedicated function.

Reply:
I implemented this and got rid of `replace_content`, although I should point out that changing the return type is technically a breaking change.

Reply:
True, but I think it's worth the version bump to avoid having two similar functions.